src/main/java/it/gov/pagopa/pu/workflow/impl/PaymentsReportingIngestionWFImpl.java

```diff
@@ -0,0 +1,99 @@
+  private String workflowQueue;
+
+  private static final int START_TO_CLOSE_TIMEOUT_IN_SECONDS = 60;
+  private static final int RETRY_INITIAL_INTERVAL_IN_MILLIS = 1000;
+  private static final double RETRY_BACKOFF_COEFFICIENT = 1.1;
+  private static final int RETRY_MAXIMUM_ATTEMPTS = 10;
+
+/** Activity stubs */
+
+  private final PaymentsReportingIngestionFlowFileActivity paymentsReportingIngestionFlowFileActivity =
+    Workflow.newActivityStub(
+      PaymentsReportingIngestionFlowFileActivity.class,
+      ActivityOptions.newBuilder()
+        .setStartToCloseTimeout(Duration.ofSeconds(START_TO_CLOSE_TIMEOUT_IN_SECONDS))
+        .setTaskQueue(workflowQueue)
+        .setRetryOptions(
+          RetryOptions.newBuilder()
+            .setInitialInterval(Duration.ofMillis(RETRY_INITIAL_INTERVAL_IN_MILLIS))
+            .setBackoffCoefficient(RETRY_BACKOFF_COEFFICIENT)
+            .setDoNotRetry(IllegalArgumentException.class.getName())
+            .setMaximumAttempts(RETRY_MAXIMUM_ATTEMPTS)
+            .build())
+        .build());
+
+
+  private final SendEmailIngestionFlowActivity sendEmailIngestionFlowActivity =
+    Workflow.newActivityStub(
+      SendEmailIngestionFlowActivity.class,
+      ActivityOptions.newBuilder()
+        .setStartToCloseTimeout(Duration.ofSeconds(START_TO_CLOSE_TIMEOUT_IN_SECONDS))
+        .setTaskQueue(workflowQueue)
+        .setRetryOptions(
+          RetryOptions.newBuilder()
+            .setInitialInterval(Duration.ofMillis(RETRY_INITIAL_INTERVAL_IN_MILLIS))
+            .setBackoffCoefficient(RETRY_BACKOFF_COEFFICIENT)
+            .setDoNotRetry(IllegalArgumentException.class.getName())
+            .setMaximumAttempts(RETRY_MAXIMUM_ATTEMPTS)
+            .build())
+        .build());
+
+  private final UpdateIngestionFlowStatusActivity updateIngestionFlowStatusActivity =
+    Workflow.newActivityStub(
+      UpdateIngestionFlowStatusActivity.class,
+      ActivityOptions.newBuilder()
+        .setStartToCloseTimeout(Duration.ofSeconds(START_TO_CLOSE_TIMEOUT_IN_SECONDS))
+        .setTaskQueue(workflowQueue)
+        .setRetryOptions(
+          RetryOptions.newBuilder()
+            .setInitialInterval(Duration.ofMillis(RETRY_INITIAL_INTERVAL_IN_MILLIS))
+            .setBackoffCoefficient(RETRY_BACKOFF_COEFFICIENT)
+            .setDoNotRetry(IllegalArgumentException.class.getName())
+            .setMaximumAttempts(RETRY_MAXIMUM_ATTEMPTS)
+            .build())
+        .build());
+
+
+  /**
+   * Workflow implementation
+   */
+
+
+  @Override
+  public void ingest(Long fileId) {
+    log.info("Ingesting file with ID: " + fileId);
+
+    PaymentsReportingIngestionFlowFileActivityResult ingestionResult =
+      paymentsReportingIngestionFlowFileActivity.processFile(fileId);
+    sendEmailIngestionFlowActivity.sendEmail("" + fileId, ingestionResult.isSuccess());
+    updateIngestionFlowStatusActivity.updateStatus(fileId, ingestionResult.isSuccess() ? "OK" : "KO");
```

Comment:
@marcomatteuccieng what's the supposed status to set for the ingestionFlowFile record here?

Moreover, Marco Cacia is adding an errorDescription on the PaymentsReportingIngestionFlowFileActivityResult DTO, what do you think if we add a techErrorDescription field on the IngestionFlowFile record to store it?